src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -111,8 +98,10 @@ private function encodeAsDollarObject(OperatorInterface $value): stdClass
     /**
      * $group stage have a specific encoding because the _id argument is required and others are variadic
      */
-    private function encodeAsGroup(GroupStage $value): stdClass
+    private function encodeAsGroup(OperatorInterface $value): stdClass
     {
+        assert($value instanceof GroupStage);
```

Comment:
Noted that this assert was moved here, it was previously handled in the `switch` statement.

Reply:
Only switch that I converted manually.